oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

```diff
@@ -1222,6 +1229,29 @@ public void removeGarbage(final VersionGCStats stats) {
 
             timer.reset().start();
             try {
+                if (embeddedVerification) {
+                    // embedded verification is done completely independent of DocumentStore
+                    // also, it's done irrespective of dry-run or not
+                    final Iterator<UpdateOp> it = updateOpList.iterator();
+                    while(it.hasNext()) {
+                        final UpdateOp update = it.next();
+                        NodeDocument oldDoc = ds.find(Collection.NODES, update.getId());
+                        if (oldDoc == null) {
```

Comment:
In case oldDoc is absent `findAndUpdate` won't do anything. So we don't need to return from here, just ignore this.

Reply:
good find, fixed in https://github.com/apache/jackrabbit-oak/pull/1262/commits/a83aee1a702b2dade42a18a37a683f0c72705fdf